web/src/app/schedules/ScheduleShiftList.js

```diff
@@ -55,115 +53,59 @@ const durString = (dur) => {
   return `${dur.days} day${dur.days > 1 ? 's' : ''}`
 }
 
-const mapQueryToProps = ({ data }) => {
-  return {
-    shifts: data.schedule.shifts.map((s) => ({
-      ...s,
-      userID: s.user.id,
-      userName: s.user.name,
-    })),
-  }
-}
-const mapPropsToQueryProps = ({ scheduleID, start, end }) => ({
-  variables: {
-    id: scheduleID,
-    start,
-    end,
+const useStyles = makeStyles({
+  datePicker: {
+    width: '100%',
   },
 })
 
-const mapStateToProps = (state) => {
-  const duration = urlParamSelector(state)('duration', 'P14D')
-  const zone = urlParamSelector(state)('tz', 'local')
-  let start = urlParamSelector(state)(
+function ScheduleShiftList({ scheduleID }) {
+  const classes = useStyles()
+
+  const [create, setCreate] = useState(null)
+  const [specifyDuration, setSpecifyDuration] = useState(false)
+  const [isClear, setIsClear] = useState(false)
+
+  const [duration, setDuration] = useURLParam('duration', 'P14D')
+  const [zone] = useURLParam('tz', 'local')
+  const [userFilter, setUserFilter] = useURLParam('userFilter', [])
+  const [activeOnly, setActiveOnly] = useURLParam('activeOnly', false)
+  const [_start, setStart] = useURLParam(
     'start',
     DateTime.fromObject({ zone }).startOf('day').toISO(),
```

Comment:
This should be using the memoized `start` value so the default doesn't keep changing (`mapStateToProps` was only evaluated once)

Reply:
Right, as long as the component is mounted, keep the default start as-is. We just don't want to change the interpretation of the URL parameter based on when a render happens, it makes for hard to detect/reproduce bugs and/or tests that fail around midnight.

Since start of day is based on the time zone though, that will need to be added as a dependency. However the `activeOnly` would be better to be just `.utc()` as there is no time of day, it just needs to be clear it's the "current timestamp".

The old code was misleading so we should fix it here -- otherwise it would seem like the second `useMemo` should track `zone` when it doesn't need it.

```js
  const defaultStart = useMemo(()=> DateTime.fromObject({ zone }).startOf('day').toISO(), [zone])
  const [_start, setStart] = useURLParam('start', defaultStart)
  const start = useMemo(
    () => (activeOnly ? DateTime.utc().toISO() : _start),
    [activeOnly, _start],
  )
```